android_17/src/product_list.js

```diff
@@ -1,28 +1,62 @@
 import React from 'react';
-import ProductData from './product_data';
-import PRODUCT_DATA from './product_data';
+
 
 export class ProductList extends React.Component {
 
+    constructor() {
+        super();
+        this.state = {
+            products: [],
+            isLoaded: false,
+        };
+    }
+
+    componentDidMount() {
+        fetch('http://localhost:5000/v1/products')
+            .then(res => res.json())
+            .then(json => {
+                this.setState({
+                    isLoaded: true,
+                    products: json['response'],
+                })
+            });
+    }
+
+
 
     render() {
+       console.log(this.state.products)
 
+        var {  isLoaded, products } = this.state;
 
+        if(!isLoaded) {
+            return <div>Loading...</div>;
+        }
+
+        else {
+        
         return (
```

Comment:
The `return` and all other lines in this `else` block should be indented
So instead of:
```
else {
return (
...
```

You have
```
else {
    return (
        ...
```